src/components/ApiKeys.vue

```diff
@@ -0,0 +1,228 @@
+<script setup lang="ts">
+import { ref } from 'vue'
+import { useI18n } from 'vue-i18n'
+import { toast } from 'vue-sonner'
+import copy from 'copy-text-to-clipboard'
+import { useSupabase } from '~/services/supabase'
+import type { Database } from '~/types/supabase.types'
+import { useMainStore } from '~/stores/main'
+import { useDisplayStore } from '~/stores/display'
+import Plus from '~icons/heroicons/plus'
+import Trash from '~icons/heroicons/trash'
+import Pencil from '~icons/heroicons/pencil'
+import Clipboard from '~icons/heroicons/clipboard-document'
+
+const { t } = useI18n()
+const displayStore = useDisplayStore()
+const main = useMainStore()
+const isLoading = ref(false)
+const supabase = useSupabase()
+const apps = ref<Database['public']['Tables']['apikeys']['Row'][]>()
+async function getKeys(retry = true): Promise<void> {
+  isLoading.value = true
+  const { data } = await supabase
+    .from('apikeys')
+    .select()
+    .eq('user_id', main.user?.id)
+  if (data && data.length)
+    apps.value = data
+
+  else if (retry && main.user?.id)
+    return getKeys(false)
+
+  isLoading.value = false
+}
+displayStore.NavTitle = ''
+getKeys()
+
+async function addNewApiKey() {
+  if (await showAddNewKeyModal())
```

Comment:
this response is true if it worked and false if not?

if this is the case, can we change it to make it more readable?
```suggestion
  const modalWorked = await showAddNewKeyModal()
  if (modalWorked)
```

note that this is only an example, feel free to change it as you think it'll be clear